### ISSUE TITLE
Libraries could not be resolved by the "cdnjs" provider

### ISSUE THREAD

**Issue by duykhiem** [cdnjs/cdnjs#14141]
Hi,

We are having an issue with the "cdnjs" provider when building our project in **both Hanoi, Vietnam, and Grimstad, Norway**.
It is happening with **both local build and Azure DevOps Pipeline build**.
**Failed on some random libraries** like jquery@3.4.1, tabulator@4.4.1, tempusdominus-bootstrap-4@5.1.2...

**Error**:

```
1:7>Project "C:\TEMP\agent\_work\1\s\Code\Insight\Insight\Bliksund.EWA.Insight.csproj" on node 2 (default targets).
     1>LibraryManagerRestore:
         
         Restore operation started
##[error]libman.json(0,0): Error LIB002: The "jquery@3.4.1" library could not be resolved by the "cdnjs" provider
     1>libman.json : error LIB002: The "jquery@3.4.1" library could not be resolved by the "cdnjs" provider [C:\TEMP\agent\_work\1\s\Code\Insight\Insight\Bliksund.EWA.Insight.csproj]
         
         One or more libraries failed to restore
         
     1>Done Building Project "C:\TEMP\agent\_work\1\s\Code\Insight\Insight\Bliksund.EWA.Insight.csproj" (default targets) -- FAILED.

Build FAILED.

       "C:\TEMP\agent\_work\1\s\Code\Insight\Insight\Bliksund.EWA.Insight.csproj" (default target) (1:7) ->
       (LibraryManagerRestore target) -> 
         **libman.json : error LIB002: The "jquery@3.4.1" library could not be resolved by the "cdnjs" provider** [C:\TEMP\agent\_work\1\s\Code\Insight\Insight\Bliksund.EWA.Insight.csproj]

```

Note: 
- We are using cdnjs as the default provider in [LibMan](https://docs.microsoft.com/en-us/aspnet/core/client-side/libman/libman-vs?view=aspnetcore-6.0)
- It was working perfectly before but s**uddenly failed today without any changes related to Libman or package references**

Could you please check and fix that asap?

Thank you very much!

**Comment by PromoFaux** [comment]
Can confirm seeing the same thing this morning with a couple of other libraries. Was working fine on Friday

![image](https://user-images.githubusercontent.com/1998970/174568657-5c3155e7-56a2-4d75-bb4f-39968d07ff71.png)

Checking on the website, they both appear to exist. ([`twitter-bootstrap@3.4.1`](https://cdnjs.com/libraries/twitter-bootstrap/3.4.1) / [`iCheck@1.0.2`](https://cdnjs.com/libraries/iCheck/1.0.2)), however I'm only being offered v5.1.3 for bootstrap (not ready to upgrade yet!) and 1.0.3 for iCheck (this is fine for me, it's a minor version)

![image](https://user-images.githubusercontent.com/1998970/174568995-01aa4764-dd92-4411-be07-f3b92115458d.png)

And now I can't build my project(s)! 🎉

**Comment by stebueh** [comment]
maybe related to https://github.com/cdnjs/cdnjs/issues/14140?

**Comment by M4ttsson** [comment]
Had the same issue. Just found that cleaning libman cache and restore the packages resolved it for me (for now). 

`dotnet tool install -g Microsoft.Web.LibraryManager.Cli`

Then go to the folder where libman.json is located and run:
`libman cache clean`
`libman restore`

https://docs.microsoft.com/en-us/aspnet/core/client-side/libman/libman-cli?view=aspnetcore-6.0

**Comment by duykhiem** [comment]
> 

Cleaning the cache helped. Thanks.

**Comment by PromoFaux** [comment]
Thanks @M4ttsson - I was about to post similar... however I am finding it to be inconsistent. Occasionally this causes new library issues to appear 🤷‍♂️, but running it several times in a row seems to have fixed it for me.

**Comment by duykhiem** [comment]
This is LibMan issue so I will close this and open new one in:
https://github.com/aspnet/LibraryManager/issues/685

**Comment by MattIPv4** [comment]
👋 For folks following this issue but not the one over in the LibMan repo, I've left a comment over there: https://github.com/aspnet/LibraryManager/issues/685#issuecomment-1160460338

Tl;dr LibMan's cdnjs integration needs a small update to how it checks and loads versions, as it is relying on a deprecated part of the API response currently that is phased out for 10% of traffic.

**Comment by MattIPv4** [comment]
The fix for LibMan has been available for a while now, and the API change has been at 100% for a week. With no further reports of issues, closing this out.

**Comment by rwb196884** [comment]
Been tearing my hair out over libman failing to restore.

Fix was to update NuGet package `Microsoft.Web.LibraryManager.Build` in the csproj that is using `libman` from version 1.1.165 to 2.1.175.

**Comment by LLIAMAH** [comment]
Problem still existing on VS2022
![image](https://user-images.githubusercontent.com/4736499/191348897-fb651e04-0c22-44ad-b8e1-007c0162c4a7.png)

Sequence of the commands and result
![image](https://user-images.githubusercontent.com/4736499/191348717-84b1166c-0a4a-413a-8c2c-be48b44aed63.png)

libman.json (left only one lib for test)
![image](https://user-images.githubusercontent.com/4736499/191349308-acde1139-9ac0-40ca-8b9e-b5eff490f50d.png)

**Comment by MattIPv4** [comment]
This is not libman support, please take up your issue with libman support if it is not working for you still. They have shipped confirmed fixes for this.

**Comment by LLIAMAH** [comment]
Ups, my bad - thought I'm in right branch :(
Too much tabs opened.

**Comment by CePur** [comment]
When I set @microsoft/signalr@latest, it resolves to the wrong version. However, @microsoft/signalr@8.0.0 resolves to the correct one. I just checked, so it shouldn't be related to  1-day cache but, using the CLI to clear and restore cache resolved the issue. Thanks for the detailed explanation and for showing the right way.
